src/AbstractSQLCompiler.ts

```diff
@@ -869,9 +878,30 @@ $$;`);
 
 		createSqlElements.push(...foreignKeys);
 		for (const index of table.indexes) {
-			createSqlElements.push(
-				index.type + '("' + index.fields.join('", "') + '")',
+			const nullsSql = index.nulls != null ? ` NULLS ${index.nulls}` : '';
+			// Non-partial indexes are added directly to the CREATE TABLE statement
+			if (index.predicate == null) {
+				createSqlElements.push(
+					index.type + nullsSql + '("' + index.fields.join('", "') + '")',
+				);
+				continue;
+			}
+			if (index.name == null) {
+				throw new Error('No name provided for partial index');
+			}
+			const comment = index.description
+				? `-- ${index.description.split(/\r?\n/).join('\n-- ')}\n`
```

Comment:
I think this could be
```suggestion
				? `-- ${index.description.replaceAll(/\r?\n/g, '\n-- ')}\n`
```
or it could be
```suggestion
				? index.description.replaceAll(/^/gm, '-- ')}\n`
```